### DISCUSSION TITLE
Timer and dimmer level

### DISCUSSION THREAD

**Discussion by TobiasF1986** [arendst/Tasmota#15591] [Q&A] [answered]
Hello,
I have a led wifi dimmer with tasmota 11.1 and would like to imitate a kind of sunrise. That's why I'd like to activate some timer with a defined dimmer value, but there is no option for that. In the menu of the timer, I only can turn the device on or off. 
E.g. I'd like to habe: 08:00 dimmer value 10%
09:00 dimmer value 20% and so on. 

Is there any possibility to do something like that on the tasmota web interface?
Thanks a lot!

**Comment by barbudor** [comment, marked as answer]
A timer can trigger a rule in which you can have any command(s)
You should also look at the Wakeup command

**Reply by TobiasF1986** [reply]
Thanks for your answer.
I tried this in the console, but it seams not to be written correctly:
Rule1 ON Time#Minute=480 DO dimmer 50 ENDON

**Reply by barbudor** [reply]
This seem to be correct but maybe you forgot to enable the rule by entering `Rule1 1` ?

Note that what I had in mind originally is to use the Timer instead of entering manually values in the rule like below:
![image](https://user-images.githubusercontent.com/2996042/167786744-a97f833c-cb3f-42e2-8e06-a7e2ca25a7a2.png)

And use the `ON Clock#Timer=1 DO ...` trigger
Which makes easy to change the time for the wakeup in the GUI instead of changing the rule or even link the trigger time to local sunrise.

You could have a look at Wakeup and Speed command to control the rampup of the light as an alternative to defining rampup point by yourself

**Reply by TobiasF1986** [reply]
Well, you are right, my mistake was not to activate Rule1 with Rule1 1. Now ot works. 

Your idea is to use the wakeup instead because I woul be limited to three rules (=> only three timers with dimmer values) I think?

**Reply by sfromis** [reply]
No, while you have 3 rule **sets**, each of those can contain many rules. Each rule is from `on` to `endon`, and with the compression taking place, you can have more than 1 KB of rule text in each set. The limitation you have with using the timer page (instead of hardcoding minutes in rules) is that you "only" have up to 16 timers, while many rules in each of the 3 sets could allow for more points in time.

Did you understand the point of using the Wakeup command? With one command you can start a whole fading-up sequence, without you having to control many points in time.

**Reply by barbudor** [reply]
Back after an IRL meeting ;) and @sfromis already re-explained everything 😂

My idea with the Timer versus Time#Minute is that it is easier to edit the start time of the sequence and, if interresting, link that to sunrise time to automatically adapt depending on the season (although using a Mem variable for the start time would help too)

**Reply by TobiasF1986** [reply]
Allright, now I do understand what you mean. 
I will try the wakeup commands at home so that I can see the reaction. I'm not at home too at the moment so that I do not see the reaction of my LED as far as I am typing something in the console via vpn connection

**Reply by TobiasF1986** [reply]
Wakeup Command works also really good. Thanks a lot for you help!!
